ieeeuoft/hackathon-template
# Move Event App Yarn Stuff

## Overview

- Resolves #241 
- Moves the package.json for Django templates and such up from the event directory, to the hackathon_site directory
- Fixes the paths in the commands as necessary
- Replaces `node-sass` with regular `sass`, because the former is apparently [deprecated](https://www.npmjs.com/package/node-sass) in favor of the new Dart version
- Updates the readme and contributing guidelines for new scss / prettier commands
- Renames the CI workflow from "event-checks" to "template-checks" (any better ideas?), and updates with the new package.json location

This is a **backwards incompatible change** with current deployments, since it will break their CI workflows


## Unit Tests Created

- Nada


## Steps to QA

- Delete the node_modules folder in your existing event folder
- Delete `hackathon_site/event/static/event/styles/css`
    - Run the local dev server, make sure styles are gone
- Run `yarn scss` or `yarn scss-watch` to make sure they pick up the files properly (check that the dev server has styles again)
- Mess up a js or scss file somewhere, and run `yarn prettier-check` to catch it, then `yarn prettier` to resolve it

As for the workflow, because we use `pull_request_target` triggers now, we just cross our fingers and hope it works once we merge.
